libpod/runtime.go

```diff
@@ -891,7 +888,6 @@ func makeRuntime(ctx context.Context, runtime *Runtime) (err error) {
 		// no containers running.  Create immediately a namespace, as
 		// we will need to access the storage.
 		if os.Geteuid() != 0 {
-			aliveLock.Unlock()
```

Comment:
I think think will cause a deadlock.  BecomeRootInUserNS will cause a re-exec, and will wait for the new process to terminate.  But that won't happen as the new process will also try to get a lock on the "alive" file

Reply:
Ouch, that sounds like deadlock. Thanks for the hint! I'll fix that up and ping after repush.

Reply:
do we need to set `locked = false` to avoid the double unlock?

Reply:
Should be better now. WDYT?

Reply:
yes, thanks for the fix.  LGTM